### DISCUSSION TITLE
Service Bus built-in reference

### DISCUSSION THREAD

**Discussion by diogonborges** [Azure/logicapps#969] [General]
Hi
Looking into : [Logic Apps Service Bus Reference](https://learn.microsoft.com/en-us/azure/logic-apps/connectors/built-in/reference/servicebus/)
specifically for https://learn.microsoft.com/en-us/azure/logic-apps/connectors/built-in/reference/servicebus/#when-messages-are-available-in-a-topic-subscription-(peek-lock)

There is no documentation about how many messages could potentially be fetched for the trigger.

This is an issue because when running a Logic App Stateless Workflow (which has no splitOn capability) a `for each` has to be used which has a configurable, but hard-coded, limit.

Looking for some extra information about the `Returns` section where it's listed 

>One or more messages received from Service Bus (peek-lock).

Which is vague and unbounded.

Thanks

**Comment by divyaswarnkar** [comment]
Hi @diogonborges - please check out this blog post and let us know if you still have any questions : https://techcommunity.microsoft.com/t5/azure-integration-services-blog/concurrency-support-for-service-bus-built-in-connector-in-logic/ba-p/4036530
